### PR TITLE
Add a test site

### PR DESCRIPTION
**Pull request by danielbrzn**
This PR seeks to add a basic test site for Markbind.

It mainly tests the includes functionality of Markbind such as:
- Nested includes
- Boilerplate includes
- Dynamic includes
- Normal includes
- Segment includes
- HTML includes

In addition, it merges the `example` folder into a unified `test_site` folder that can be used for testing.

### CODE REVIEW

**Comment by nicholaschuayunzhi** [comment]
I have also managed to find a way to do automated testing with Travis when submitting PRs.

https://github.com/nicholaschuayunzhi/markbind-cli/pull/1

It tests for for linting error, and then a file compare between an expected site and a generated site during the test.

EDIT: Should make a new pr or commit on top of this?

### Considerations
- Need to build expected site on unix system due to path separators
- Need to set up travis on this repo
- Currently there is no automated test for windows users locally

**Comment by Gisonrg** [comment]
~~Can you send a separate PR to delete the outdated `example` folder? Thanks :D~~

Ignore this, didn't read the description carefully

**Comment by Gisonrg** [comment]
@nicholaschuayunzhi 
I think the test site Daniel created could be used in the integration test, a.k.a end-to-end test including system IO to simulate a real site building process. 
While your work (I have seen all the mock objects, good work 👍) could be used as unit test, which provides a fast way to ensure things are work / validate functions are called, etc.
Both of your works would be very helpful in testing 💯

**Comment by rachx** [comment]
Can/Should we include known bugs in the test site (e.g. multiple inclusion of a modal)?
It might be easier for testing and debugging (we can all refer to the same copy without creating our own test case) and it can be kept in the future for regression testing.

**Comment by acjh** [comment]
@rachx Yes, and:

> - We should differentiate between "issues" (reproducing bugs) and "PRs" (regression testing).

— https://github.com/MarkBind/markbind/issues/121#issuecomment-359710181

**Comment by rachx** [comment]
It will be good to add examples of the supported markdown syntax (radio buttons, code blocks etc) and components (question, popover) etc but that can be done in a separate issue.

Hopefully we can merge this soon as it makes testing more convenient

File: test_site/sub_site/site.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "baseUrl": "/sub_site",
+  "pages": [
+    {
+      "src": "index.md",
+      "title": "Hello World"
+    }
+  ],
+  "ignore": [
+    "_site/*",
+    "site.json"
```
Comment (nicholaschuayunzhi) [inline test_site/sub_site/site.json]: could we ignore `*md` files as well? I believe our test does not need to check for copied md files

**Comment by nicholaschuayunzhi** [comment]
I have updated the script to diff html differently with a separate script, ignoring path separators.
https://github.com/nicholaschuayunzhi/markbind-cli/pull/2

File: test_site/sub_site/site.json (hunk 1 above)
Comment (danielbrzn) [reply test_site/sub_site/site.json]: Sure, will add the required changes.

**Review by acjh — APPROVED** [review]

**Comment by acjh** [comment]
- Let's move this into **test/test_site**, so that the unit tests can also be in the same folder in future.
  @nicholaschuayunzhi This will affect the paths in your PRs for *automated testing* and *unit tests*.

- Let's add this to **.gitignore**:
  ```
  # Generated site (MarkBind)
  test/test_site/_site
  ```

**Comment by acjh** [comment]
@danielbrzn In case you missed my comment above.

**Comment by danielbrzn** [comment]
Sorry about that, will get on it.

**Review by Gisonrg — APPROVED** [review]
Good work!

**Pull request merged**